### ISSUE TITLE
Feature request: Add option for dekstop icons not to overlap each other

### ISSUE THREAD

**Issue by ThanosApostolou** [mate-desktop/caja#644]
Currently caja/mate-desktop (and nautilus/gnome, nemo/cinnamon) are allowing desktop icons to overlap each other even with the option `Keep Aligned` enabled. On the contrary thunar/xfdesktop make a perfect and really beautiful alignment of desktop icons. See difference at pictures:

MATE:
https://ubuntu-mate.community/uploads/default/original/2X/2/21613bab11e6f5364276c362dc0b7a1f4993de99.png

XFCE:
https://ubuntu-mate.community/uploads/default/original/2X/1/12150ed1dfc27742863d65b1fcb97a9457975d1e.png

We had a discussion about proper alignment of desktop icons at https://ubuntu-mate.community/t/proper-allignement-of-desktop-symbols/9153 and we thought that it would be really useful to add an option so that icons don't overlap each other (even just a dconf/gsettings option would be enough). This is really important especially for users that come from Windows (even xfce since it's currently not under a lot development, so, more and more users will switch to MATE in the future).

**Comment by kn00tcn** [comment]
i dont think it needs to be an option, the current way is simply wrong https://github.com/linuxmint/nemo/issues/108

the grid is too small, what we need is an option to change the grid distance

finally, the good way to handle arbitrary aspect image thumbnails is to constrain & vert+hor center align them into a standard icon size, so ALL icons should be the same size on the same fixed grid, problem solved

this is important for any desktop computing, nothing to do with people coming from windows

**Comment by lukefromdc** [comment]
Fixing this right would also fix one of two problems blocking https://github.com/mate-desktop/caja/pull/635 , the other being that auto-moved icon positions are not saved.

**Comment by dryphi** [comment]
Agree please fix. Overlapping icons still a problem in Ubuntu 18.04

**Comment by SlateDave** [comment]
Why is fixing so hard? Its so annoying and still hasn't been fixed in years...

**Comment by lukefromdc** [comment]
As I recall, he lower level code that attaches information to the icons does not include certain key variables, so fixing the saving of icon positions would be a major refactoring

**Comment by FinixFighter** [comment]
Any progress on this issue?

**Comment by harlekin777** [comment]
In 2017 this issue was solved in Nemo: https://github.com/linuxmint/nemo/issues/108 
kn00tcn mentioned the discussion in Nemo above earlier in 2016, before the solution appeared over there.
You may check it and see how the problem was fixed in Nemo. It may give you an idea how to do it in Caja. It would be great to solve this problem in Caja as well; it is really disturbing for many years now.

**Comment by alkisg** [comment]
This is indeed rather annoying.
I'm sure most people will agree, even if they won't spend the time to locate and comment on this issue; it took me more than 5 years of being annoyed to decide to search for it! :D

**Comment by lukefromdc** [comment]
I found this one to be beyond my skills back in 2016 due to the icons themselves not having their positions stored in the structures associated with them. I had tried to fix another but similar issue but there was no way to get the positions to keep

**Comment by alkisg** [comment]
The [Nemo solution mentioned above](https://github.com/mate-desktop/caja/issues/644#issuecomment-982620221) doesn't apply for Caja?

**Comment by lukefromdc** [comment]
None of those links here (not ONE of them) takes me to actual Nemo commits.
